libsmb2 mingw32 compilation/usage

Hi,

I have been trying to use libsmb2 with our my project that is based on x86_64-w64-mingw32-gcc.

I was able to solve a few errors, but i reached errors I can't find much information about on the internet. Linking both ws2_32 and ucrt solved some of the undefines.

Command:
`x86_64-w64-mingw32-gcc -Wall test.c -IInclude -L lib -lsmb2 -lws2_32 -lucrt -o mytest`

code

```
#include <stdint.h>
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
#include "smb2/smb2.h"
#include "smb2/libsmb2.h"
#include "smb2/libsmb2-raw.h"

int main()
{
    struct smb2_context* smb2;
    smb2 = smb2_init_context();
}
```


errors:


```
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
/root/Documents/myGitProjects/smbclient/x86_64-w64-mingw32-cross/bin/../lib/gcc/x86_64-w64-mingw32/11.2.1/../../../../x86_64-w64-mingw32/bin/ld: lib/libsmb2.lib(x64/Release/init.obj):(.text$mn+0x7c): undefined reference to `snprintf'
/root/Documents/myGitProjects/smbclient/x86_64-w64-mingw32-cross/bin/../lib/gcc/x86_64-w64-mingw32/11.2.1/../../../../x86_64-w64-mingw32/bin/ld: lib/libsmb2.lib(x64/Release/socket.obj):(.text$mn+0x16): undefined reference to `__chkstk'
/root/Documents/myGitProjects/smbclient/x86_64-w64-mingw32-cross/bin/../lib/gcc/x86_64-w64-mingw32/11.2.1/../../../../x86_64-w64-mingw32/bin/ld: lib/libsmb2.lib(x64/Release/socket.obj):(.text$mn+0x12): undefined reference to `__chkstk'
collect2: error: ld returned 1 exit status
```

the 3 main errors i get are related to undefined references:
```
undefined reference to `snprintf'
undefined reference to `__chkstk'
undefined reference to `__chkstk'
```

--- a/lib/compat.h
+++ b/lib/compat.h
@@ -23,7 +23,7 @@
 extern "C" {
 #endif
 
-#if defined(_XBOX) || defined(_WINDOWS)
+#if defined(_XBOX) || defined(_WINDOWS) || defined(__MINGW32__)
 
 #ifdef _XBOX
 /* XBOX Defs begin */
@@ -101,7 +101,7 @@ typedef int t_socket;
 #define EBADF WSAENOTSOCK
 #endif
 
-#ifdef _XBOX
+#if defined(_XBOX) || defined(__MINGW32__)
 #define snprintf _snprintf
 #endif
 

